app/controllers/sns_controller.rb

```diff
@@ -0,0 +1,53 @@
+# These endpoints allow the Amazon SES service to notify us when there are bounced
+# emails or complaints (ex we are marked as spam). The expected JSON is particular to
+# Amazon's SNS notification service, and thus will not work for other services.
+class SnsController < ApplicationController
+  protect_from_forgery with: :null_session
+  before_filter :verify_amazon_authorize_key
+
+  def bounce
+    begin
+      success = true
+      message = JSON.parse(request.body.read)['Message']
+      bounce_arr = JSON.parse(message)['bounce']['bouncedRecipients']
+      bounce_arr.each do |recipient|
+        Bounce.create(email: recipient['emailAddress'].downcase)
+      end
+    rescue
+      success = false
+    ensure
+      render json: {success: success}
+    end
+  end
+
+  def complaint
+    logger.info 'Received Amazon SES complaint notification'
+    begin
+      success = true
+      message = JSON.parse(request.body.read)['Message']
+      complaint = JSON.parse(message)['complaint']
+      if complaint.nil?
+        # Save non-standard requests, ex subscription confirmation
```

Comment:
Yeah, they go to the same endpoint that the complaint notifications will eventually go to. 

I was thinking if the format changes for some reason it might be nice to still capture what amazon sends us - posts from anyone else won't make it to this controller because they won't have the aws key. I suppose the alternative would be to just log the confirmation url - there should only be one.